src/lib/utils/useScreenReaderTracking.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react"
+import { AccessibilityInfo } from "react-native"
+import { postEventToProviders } from "./track/providers"
+
+// the purpose of this hook is to track the screen reader status of the user's device
+export const useScreenReaderTracking = () => {
+  const [isScreenReaderEnabled, setIsScreenReaderEnabled] = useState(false)
+  useEffect(() => {
+    const getScreenReaderStatus = async () => {
+      const isEnabled = await AccessibilityInfo.isScreenReaderEnabled()
+      setIsScreenReaderEnabled(isEnabled)
```

Comment:
**thought:** I am not sure why we need to use the state here. Can't we maybe do
```suggestion
      postEventToProviders(tracks.trackScreenReaderStatus(isEnabled))
```
and not use a state

Reply:
Good point. The initial approach was without `useState`. The reason I used state here is because if I don't (and my code being like this):
```suggestion
  export const useScreenReaderTracking = () => {
    useEffect(() => {
      const getScreenReaderStatus = async () => {
        const isEnabled = await AccessibilityInfo.isScreenReaderEnabled()
        return isEnabled
      }

      getScreenReaderStatus()
      postEventToProviders(tracks.trackScreenReaderStatus(getScreenReaderStatus()))
    }, [])
  }
```
I get the following ts error `Argument of type 'Promise<boolean>' is not assignable to parameter of type 'boolean'.ts(2345)`

Reply:
ah yeah. it's ts sense/non-sense  but I guess it's understandable in case the promise fails

Reply:
I agree with getting rid of the state.

maybe try to move the thing in a try-catch, so if you get a boolean you postevent, if you get an error you send nothing or send a "N/A" or something.